src/components/figures/chart/scorecard/chart_scorecard.ts

```diff
@@ -1,294 +1,51 @@
-  }
-
-  /** Return the element with he widest text in the chart */
-  private getWidestElement(): ScorecardScalableElement {
-    const baseline = new BaselineElement(this.runtime.baselineStyle);
-    const keyValue = new KeyValueElement(this.runtime.keyValueStyle);
-
-    return baseline.getElementWidth(BASELINE_BOX_HEIGHT_RATIO, this.ctx, this) >
-      keyValue.getElementWidth(KEY_BOX_HEIGHT_RATIO, this.ctx, this)
-      ? baseline
-      : keyValue;
-  }
-}
-
-abstract class ScorecardScalableElement {
-  constructor(private style: Style = {}) {}
-
-  /** Return the width of an scorecard element in pixels */
-  abstract getElementWidth(
-    fontSize: number,
-    ctx: CanvasRenderingContext2D,
-    chart: ScorecardChart
-  ): Pixel;
-
-  /**
-   * Get the maximal height of an element of the scorecard.
-   *
-   * This is computed such as all the height is taken by the elements, even if there is no title or baseline.
-   */
-  abstract getElementMaxFontSize(availableHeight: Pixel, chart: ScorecardChart): number;
-
-  protected measureTextWidth(ctx: CanvasRenderingContext2D, text: string, fontSize: number) {
-    const italic = this.style.italic ? "italic" : "";
-    const weight = this.style.bold ? "bold" : "";
-    ctx.font = `${italic} ${weight} ${fontSize}px ${DEFAULT_FONT}`;
-    return ctx.measureText(text).width;
-  }
-}
-
-class BaselineElement extends ScorecardScalableElement {
-  getElementWidth(fontSize: number, ctx: CanvasRenderingContext2D, chart: ScorecardChart): Pixel {
-    if (!chart.runtime) return 0;
-    const baselineStr = chart.baseline;
-    // Put mock text to simulate the width of the up/down arrow
-    const largeText = chart.baselineArrowDirection !== "neutral" ? "A " + baselineStr : baselineStr;
-    let textWidth = this.measureTextWidth(ctx, largeText, fontSize);
-    // Baseline descr font size should be smaller than baseline font size
-    textWidth += this.measureTextWidth(
-      ctx,
-      chart.baselineDescr,
-      fontSize * BASELINE_DESCR_FONT_RATIO
-    );
-    return textWidth;
-  }
-
-  getElementMaxFontSize(availableHeight: Pixel, chart: ScorecardChart): number {
-    if (!chart.runtime) return 0;
-    const haveBaseline = chart.baseline !== "" || chart.baselineDescr;
-    const maxHeight = haveBaseline ? BASELINE_BOX_HEIGHT_RATIO * availableHeight : 0;
-    return maxHeight / LINE_HEIGHT;
-  }
-}
-
-class KeyValueElement extends ScorecardScalableElement {
-  getElementWidth(fontSize: number, ctx: CanvasRenderingContext2D, chart: ScorecardChart): Pixel {
-    if (!chart.runtime) return 0;
-    const str = chart.keyValue || "";
-    return this.measureTextWidth(ctx, str, fontSize);
+    let previousRuntime = this.runtime;
+    let previousWidth = this.props.figure.width;
+    let previousHeight = this.props.figure.height;
+    onPatched(() => {
+      const runtime = this.runtime;
+      const figure = this.props.figure;
+      if (
+        deepEquals(runtime, previousRuntime) &&
```

Comment:
I think we can compare by reference instead of a deep comparison.
It would allow to use `useEffect` which does the comparison (by reference) for us.
The entire `onPatch` and `onMounted` could be replaced by:
```ts
useEffect(
      this.createChart.bind(this),
      () => [this.runtime, this.props.figure.width, this.props.figure.height, this.canvas.el]
);
```
Note that technically, `this.canvas.el` is also a dependency